pallets/multisig/src/lib.rs

```diff
@@ -136,106 +131,95 @@ fn with_base_weight(
-
-        /// The last transaction version, used for `on_runtime_upgrade`.
-        TransactionVersion get(fn transaction_version) config(): u32;
-        /// Storage version.
-        StorageVersion get(fn storage_version) build(|_| Version::new(3)): Version;
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + IdentityConfig {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Weight information for extrinsics in the multisig pallet.
+        type WeightInfo: WeightInfo;
+        /// Maximum number of signers that can be added/removed in one call.
+        #[pallet::constant]
+        type MaxSigners: Get<u32>;
     }
-}
-
-decl_module! {
-    /// A multisig module.
-    pub struct Module<T: Config> for enum Call where origin: T::RuntimeOrigin {
-        type Error = Error<T>;
 
-        fn deposit_event() = default;
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
         fn on_runtime_upgrade() -> Weight {
             use sp_version::RuntimeVersion;
-            use polymesh_primitives::migrate::kill_item;
             let mut weight = Weight::zero();
 
-            // Kill pending proposals if the transaction version is upgraded
+            // Clear pending proposals if the transaction version is upgraded
             let current_version = <T::Version as Get<RuntimeVersion>>::get().transaction_version;
-            if TransactionVersion::get() < current_version {
-                TransactionVersion::set(current_version);
-                // TODO: Replace this code with `Proposal*::remove*` calls.
-                // Doing so will provide compile-time checks.  The current code
-                // will fail silently if storage names changes.
-                for item in &["Proposals", "ProposalIds", "ProposalDetail", "Votes"] {
-                    kill_item(NAME, item.as_bytes())
-                }
-                // TODO: Improve this weight.
-                weight.saturating_accrue(DbWeight::get().reads_writes(4, 4));
+            if TransactionVersion::<T>::get() < current_version {
+                TransactionVersion::<T>::set(current_version);
+                // TODO: Multiblock migration.
+                let mut removed = 0;
+                let res = Proposals::<T>::clear(u32::max_value(), None);
+                removed += res.unique;
+                let res = ProposalIds::<T>::clear(u32::max_value(), None);
+                removed += res.unique;
+                let res = ProposalVoteCounts::<T>::clear(u32::max_value(), None);
```

Comment:
Why do we have to clear this new storage? Shouldn't we clear `ProposalDetail`?

Reply:
This code is for future chain upgrades that change the transaction version (breaking changes).  The old storage is cleared below in the migration code.